include/seqan3/search/dream_index/detail/bitvector.hpp

```diff
@@ -0,0 +1,1194 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \brief Provides seqan3::detail::bitvector.
+ */
+
+#pragma once
+
+#include <sdsl/bit_vectors.hpp>
+
+#include <seqan3/core/concept/cereal.hpp>
+#include <seqan3/std/iterator>
+
+namespace seqan3::detail
+{
+
+//!\cond
+// Forward declaration
+class bitvector_compressed;
+//!\endcond
+
+/*!\brief A wrapper for the sdsl::bit_vector.
+ * \implements std::ReservableContainer
+ *
+ * This class additionally provides get_int(), set_int() and a custom
+ * \ref resize(size_type bits_, size_type block_size, size_type block_size_) "resize" function.
+ */
+class bitvector_uncompressed
+{
+private:
+    // friend bitvector_compressed;
```

Comment:
?

Reply:
I'll change it. But basically we need to befriend the compressed one because it needs access (compressed is only constructible through a uncompressed one).